PC/getpathp.c

```diff
@@ -537,14 +537,25 @@ get_program_full_path(const PyConfig *config,
     wchar_t program_full_path[MAXPATHLEN+1];
     memset(program_full_path, 0, sizeof(program_full_path));
 
+    if (!GetModuleFileNameW(NULL, program_full_path, MAXPATHLEN)) {
+        /* GetModuleFileName should never fail when passed NULL */
+        return _PyStatus_ERR("Cannot determine program path");
+    }
+
     /* The launcher may need to force the executable path to a
      * different environment, so override it here. */
     pyvenv_launcher = _wgetenv(L"__PYVENV_LAUNCHER__");
     if (pyvenv_launcher && pyvenv_launcher[0]) {
+        /* If overridden, preserve the original full path */
```

Comment:
It seems like macOS code should also be updated in config_init_program_name(), but that can be done later.

Note: I'm not sure if venv() code of Lib/site.py is redundant or not.

Reply:
Honestly, I can't figure out exactly what the macOS logic is supposed to be... I don't think it's the same as what I have for Windows, even though theoretically it is fixing the same kind of issue.